Sources/NIOResumableUploadDemo/main.swift

```diff
@@ -0,0 +1,114 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import HTTPTypes
+import NIOCore
+import NIOHTTP1
+import NIOHTTPTypes
+import NIOHTTPTypesHTTP1
+import NIOPosix
+import NIOResumableUpload
+import System
+
+@available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
+final class UploadServerHandler: ChannelDuplexHandler {
+    typealias InboundIn = HTTPRequestPart
+    typealias OutboundIn = Never
+    typealias OutboundOut = HTTPResponsePart
+
+    let directory: FilePath
+    var fileHandle: FileHandle? = nil
+
+    init(directory: FilePath) {
+        self.directory = directory
+    }
+
+    func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+        switch self.unwrapInboundIn(data) {
+        case .head(let request):
+            switch request.method {
+            case .post, .put:
+                if let requestPath = request.path {
+                    let path = self.directory.appending(requestPath)
```

Comment:
This might be a bad idea since people can send `../` in the request path

Reply:
I'm now using `standardized` URL and comparing the prefix which hopefully addresses the issue